src/mds/SessionMap.cc

```diff
@@ -1216,6 +1216,13 @@ int SessionFilter::parse(
       state = v;
     } else if (k == "id") {
       std::string err;
+      if (v == "*") {
+        // evict all clients , by default id set to 0
```

Comment:
So, is this allowing all clients to be evicted by passing in a `*` and not `0` in filter?

Reply:
Yes, `id=*` will evict all clients now and for `id=0` we will be seeing `EINVAL` error.

Reply:
I didn't get why both should lead to same result. for `id=*` it should evict all clients as we are considering `*` for all whereas `id=0` should be restricted from the user along with throwing error. I guess this was expected from the suggestions above.

Reply:
Didn't id=0 evicting all client as mentioned by @zhsgao ?

Reply:
Referring this https://github.com/ceph/ceph/pull/49974#issuecomment-1791828872 . It was decided that we will be adding `id=*` for eviction of all clients and `id=0` which earlier evicts all clients, should be disallowed, right? This is being done in the current patch.

Reply:
OK. I was under the impression to disallow evicting all client en-masse. I'm fine with what this currently implements since passing `*` is obvious to the user to evict all clients and id=0 is blocked. Thanks for explaining @neesingh-rh